src/rp2_common/pico_unique_id/unique_id.c

```diff
@@ -26,3 +26,16 @@ static void __attribute__((constructor)) _retrieve_unique_id_on_boot(void) {
 void pico_get_unique_board_id(pico_unique_board_id_t *id_out) {
     *id_out = retrieved_id;
 }
+
+void pico_get_unique_board_id_string(char *id_out, size_t len) {
```

Comment:
also for better or worse we've tended to use `uint` rather than `size_t` for sizes in APIs

Reply:
All done.  To preserve hex ordering, we actually need to shift `(4 - 4 * (i&1))` or the hex nibbles end up backwards.

Reply:
It's a good thing that `*` has a higher precedence than `-` :wink: